server/src/test/java/io/druid/server/initialization/JettyTest.java

```diff
@@ -360,4 +423,33 @@ public Response exception(
       throw new IOException();
     }
   }
+
+  public static class DummyAuthFilter implements Filter {
```

Comment:
If it is just for reference implementation, then the unit test here should be good enough. It is doing everything one needs to do in their extension (creating and binding ServletFilterHolder impl, creating sample filter etc). Moving it to main code from test wouldn't change anything I guess.